exercises/concept/go/src/Go.elm

```diff
@@ -0,0 +1,40 @@
+module Go exposing (..)
+
+
+type Player
+    = Black
+    | White
+
+
+type alias Game =
+    { whiteCapturedStones : Int
+    , blackCapturedStones : Int
+    , player : Player
+    , error : String
+    }
+
+
+type alias Rule =
+    Game -> Result String Game
+
+
+type alias NonValidatingRule =
+    Game -> Game
+
+
+changePlayer : Game -> Game
+changePlayer game =
+    let
+        newPlayer =
+            if game.player == White then
+                Black
+
+            else
+                White
+    in
+    { game | player = newPlayer }
```

Comment:
There is a slight risk that the student will start editing these definitions and function. One option to consider is to put this code in a second `editor` file (which are shown as read-only in the online editor, see [the docs](https://exercism.org/docs/building/tracks/concept-exercises#h-file-meta-config-json)).

This would also allow you to move the implementation of the rules from the tests file to the editor file.

Reply:
This is a good idea and I've done it, although the implementation of the rules is a test only thing, so I've left them where they are.